### PR TITLE
fix(background): prevent monetization at prerendering page

### PR DESCRIPTION
**Pull request by dianafulga**
#477

### CODE REVIEW

**Comment by dianafulga** [comment]
I thought that we can still save the links at prerender, but the icon and the monetization should not start until actual navigation to the page.

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">b8f6a3cb375c9c78752ed3041d6372a2efe3fbc5</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/10261260645">Run #10261260645</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/26841319514/artifacts/1779511607">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/26841319514/artifacts/1779511609">Download</a></td></tr>
</tbody>
</table>

File: src/background/services/monetization.ts [hunk 1]
```diff
@@ -95,6 +95,8 @@ export class MonetizationService {
       sessions.set(requestId, session)
     })
 
+    if (isPrerendering) return;
+
     this.events.emit('monetization.state_update', tabId)
```
Comment (dianafulga) [inline src/background/services/monetization.ts:100]: I would say this one should be emitted only after navigating to the page

File: src/background/utils.ts [hunk 2]
```diff
@@ -88,8 +88,10 @@ export const getTab = (sender: Runtime.MessageSender): Tabs.Tab => {
 export const getSender = (sender: Runtime.MessageSender) => {
   const tabId = getTabId(sender)
   const frameId = notNullOrUndef(sender.frameId, 'sender.frameId')
+  const isPrerendering = (sender as any).documentLifecycle === 'prerender' && sender.frameId !== 0;
```
Comment (dianafulga) [inline src/background/utils.ts:91]: the polyfill is missing a few attributes

**Comment by raducristianpopa** [comment]
Closed in favor of #486.

**Pull request closed without merging**